apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -501,12 +526,47 @@ export const resourceRouter = router({
         resourceId,
       })
 
-      const result = await db
-        .deleteFrom("Resource")
-        .where("Resource.id", "=", String(resourceId))
-        .where("Resource.siteId", "=", siteId)
-        .where("Resource.type", "!=", ResourceType.RootPage)
-        .executeTakeFirst()
+      const result = await db.transaction().execute(async (tx) => {
+        const user = await db
```

Comment:
issue: use `tx` instead of `db`.

Reply:
shifting it out instead, think shouldn't be part of all or nth tx